coalib/bearlib/languages/documentation/DocumentationComment.py

```diff
@@ -33,3 +38,93 @@ def __init__(self, documentation, language, docstyle,
 
     def __str__(self):
         return self.documentation
+
+    def parse_documentation(self):
+        """
+        Parses documentation independent of language and docstyle.
+
+        :return:
+            A list of the parsed documentation metadata.
+        :raises:
+            NotImplementedError
+        """
+        if self.language == "python" and self.docstyle == "default":
+            return self.parse_python_default()
+        else:
+            raise NotImplementedError(
+                "Documentation parsing for {0.language!r} in {0.docstyle!r}"
+                " has not been implemented yet".format(self))
+
+    def parse_python_default(self):
+        """
+        Parse documentation. Usable attributes are:
+
+        - ``:param``
+        - ``:return:``
+
+        :return: A list of parsed sections(descriptions, params, return values)
+        """
+        lines = self.documentation.splitlines(keepends=True)
+
+        param_identifier = (":param ", ": ")
+        return_identifier = ":return:"
+
+        return self._parse_documentation_with_symbols(lines, param_identifier,
+                                                      return_identifier)
```

Comment:
Well it would be nice if I could explain the documentation pattern of every language + docstyle in its respecitive function docstring?

For example for `parse_python_default` I explain python docstrings and for `c_python_doxygen` I explain how c documentation is written in doxygen..